crates/pallet-domains/src/lib.rs

```diff
@@ -3098,3 +3106,8 @@ pub fn calculate_tx_range(
     };
     new_tx_range.clamp(lower_bound, upper_bound)
 }
+
+/// An abstraction which allows pallet-domains to get the block timestamp from pallet-timestamp
+pub trait BlockTimestampProvider {
```

Comment:
I believe this trait is not required. Timestamp primitives already provide an abstraction for this

Reply:
Specifically, https://docs.rs/frame-support/38.0.0/frame_support/traits/trait.Time.html